File: packages/jsii-dotnet-runtime/src/AWS.Jsii.Runtime/JsiiException.cs
using AWS.Jsii.JsonModel.Api.Response;
using System;

namespace AWS.Jsii.Runtime
{
    public class JsiiException : Exception
    {
        public ErrorResponse ErrorResponse { get; }

        public JsiiException(string message, Exception innerException)
            : base(message, innerException)
        {
        }

        public JsiiException(ErrorResponse response, Exception innerException)
            : base(response.Error, innerException)
        {
            ErrorResponse = response;
        }
    }
}

File: packages/@jsii/java-runtime/project/src/main/java/software/amazon/jsii/JsiiException.java
package software.amazon.jsii;

/*
 * An error raised by the jsii runtime.
 */
public abstract class JsiiException extends RuntimeException {
    public static final long serialVersionUID = 1L;

    public static enum Type {
        JSII_FAULT("@jsii/kernel.Fault"),
        RUNTIME_EXCEPTION("@jsii/kernel.RuntimeException");

        private final String errorType;

        Type(String str) {
            this.errorType = str;
        }

        public String toString() {
            return this.errorType;
        }
    }

  /**
   * Creates an exception.
   * @param message The error message
   */
  JsiiException(final String message) {
      super(message);
  }

  /**
   * Creates an exception.
   * @param e The error that caused this exception
   */
  JsiiException(final Throwable e) {
      super(e);
  }

  /**
   * Creates an exception.
   * @param message The error message
   * @param e The error that caused this exception
   */
  JsiiException(final String message, final Throwable e) {
      super(message, e);
  }
}
